src/Symfony/Component/Security/Core/Authentication/Token/OfflineTokenInterface.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Authentication\Token;
+
+/**
+ * Interface used for marking tokens that do not represent the currently logged-in user.
+ *
+ * @author Nate Wiebe <nate@northern.co>
+ */
+interface OfflineTokenInterface extends TokenInterface
```

Comment:
the word offline is weird as first PR read

- AnonymousToken?
- UnknownToken?
- UnawareToken?

Reply:
I feel like offline is still the most understandable word here, but open to suggestions. Token in my head is the representation of the user, and in this context they would be offline as far as the app is concerned. I could maybe see something like `SessionlessTokenInterface` or `StatelessTokenInterface`, but then that might get confused with stateless firewalls, which is different.

Reply:
`RuntimeTokenInterface`?

Reply:
I'm personally fine with `OfflineTokenInterface`

Reply:
`OfflineTokenInterface` implies no connection at all, which could be wrong.
Use case: an admin logged in during a request, send mails synchronously to a list of users, and the email template displays links depending on permissions of the recipient user.

Reply:
- RetrieveOnlyToken / ProvideOnlyToken
- RetrieverToken
- QueryToken (which could be a bit misleading)
- HolderToken
- DifferentUserToken

Reply:
How about one of these?
- StatelessToken (I'm leaning towards this one if we change it)
- SessionlessToken
- UserRepresentationToken

Reply:
SessionlessToken sounds best to me.

Reply:
SessionLessToken and StatelessToken are both confusing, because the token of the current user stored in the TokenStorage might also be session-less (if your configure your firewall as stateless, it does not store the current token in the session).
Having 2 different meaning for `stateless` inside the component is a bad idea IMO.

I would also keep `OfflineTokenInterface` (it might not be a perfect name, but I don't have a better idea, and I still prefer it over other names suggested here).